### PR TITLE
DROOLS-3754: [DMN Designer] Data Types - Constraints (Range/Enumeration) - Add 'Day/Time Duration' component when the type is 'Day/Time Duration'

### PR DESCRIPTION
**Pull request by karreiro**
See:
- https://issues.jboss.org/browse/DROOLS-3754
- https://issues.jboss.org/browse/DROOLS-3760

---

demo

---

Part of an ensemble:
- https://github.com/kiegroup/appformer/pull/668
- https://github.com/kiegroup/kie-wb-common/pull/2548

### CODE REVIEW

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/listview/constraint/common/typed/day/time/DayTimeValue.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kie.workbench.common.dmn.client.editors.types.listview.constraint.common.typed.day.time;
+
+class DayTimeValue {
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/listview/constraint/common/typed/day/time/DayTimeValue.java]: I have feeling we don't have to need introduce this class, if we expose some more methods of `MomentDuration`

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/listview/constraint/common/typed/day/time/DayTimeValueConverter.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+                                            final int hours,
+                                            final int minutes,
+                                            final int seconds) {
+        final JSONObject jsonObject = new JSONObject();
+        jsonObject.put("days", number(days));
+        jsonObject.put("hours", number(hours));
+        jsonObject.put("minutes", number(minutes));
+        jsonObject.put("seconds", number(seconds));
+        return jsonObject.getJavaScriptObject();
+    }
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/listview/constraint/common/typed/day/time/DayTimeValueConverter.java]: This is related to my previous comment, if we enable creation of `MomentDuration` directly there, I think we wouldn't need `DayTimeValue`.

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/listview/constraint/common/typed/day/time/DayTimeValue.java (hunk 1 above)
Comment (karreiro) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/listview/constraint/common/typed/day/time/DayTimeValue.java]: Thank you for the comment, @jomarko. However, I created the `DayTimeValue` to not couple any class from the `DayTimeSelector` (expect the `DayTimeValueConverter`)  with the `moment.js`. So, I think we could keep it as is :-)

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/listview/constraint/common/typed/day/time/DayTimeValueConverter.java (hunk 2 above)
Comment (karreiro) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/listview/constraint/common/typed/day/time/DayTimeValueConverter.java]: I've commented about this here ~>
https://github.com/kiegroup/kie-wb-common/pull/2548#discussion_r267895361 ;-)

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/listview/constraint/common/typed/common/MinMaxValueHelper.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+    private static boolean onFocusOut(final HTMLInputElement input) {
+        return onKeyUp(input);
+    }
+
+    private static boolean isValidValue(final HTMLInputElement input,
+                                        final int inputValue) {
+
+        final int max = toInt(input.max, MAX_VALUE);
+        final int min = toInt(input.min, MIN_VALUE);
```
[the remark this answers is not in the document]
Comment (karreiro) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/listview/constraint/common/typed/common/MinMaxValueHelper.java]: Fixed. Thanks! :-)

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/listview/constraint/common/typed/common/MinMaxValueHelper.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        final String inputValue = input.value;
+
+        if (isValidValue(input, toInt(inputValue))) {
+            setOldAttribute(input, inputValue);
+        }
+
+        return true;
+    }
+
+    static boolean onKeyUp(final HTMLInputElement input) {
```
[the remark this answers is not in the document]
Comment (karreiro) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/listview/constraint/common/typed/common/MinMaxValueHelper.java]: Fixed. Thanks! :-)

**Comment by karreiro** [comment]
@jomarko Thank you for the review and for spotting the issue in the video (very good catch).

The fix and the proposed changes were applied in the new commit.

**Comment by karreiro** [comment]
@danielzhe Changes applied. I've fixed the issue you mentioned on GChat, see:

**Before:**
range

**After:**

**Pull request merged**